### PR TITLE
julia: 1.8.0-rc1

### PR DESCRIPTION
**Pull request by haampie**
I'm giving up on trying to add patches to libuv and instead I'd just add
libuv-julia; I don't think we need a virtual dependency here, since julia can't
depend on both libuv and libuv-julia.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libuv-julia/package.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class LibuvJulia(AutotoolsPackage):
+    """Multi-platform library with a focus on asynchronous IO"""
+    homepage = "https://libuv.org"
+    url      = "https://github.com/JuliaLang/libuv/archive/refs/heads/julia-uv2-1.44.1.tar.gz"
+
+    version('1.44.1', sha256='f931e7825702cbb6d07486d92e5436990cf20f91e2b56d6f759822c0f832b13e')
```
Comment (giordano) [inline var/spack/repos/builtin/packages/libuv-julia/package.py]: I'm not sure this is a stable reference, there have been pushes to this branch recently, after it had been prepared for use in the Julia release, for example https://github.com/JuliaLang/libuv/pull/21

**Review by vchuravy — DISMISSED** [review]

File: var/spack/repos/builtin/packages/julia/package.py [hunk 2]
```diff
@@ -26,10 +26,11 @@ class Julia(MakefilePackage):
     maintainers = ['glennpj', 'vchuravy', 'haampie']
 
     version('master', branch='master')
-    version('1.7.2', sha256='0847943dd65001f3322b00c7dc4e12f56e70e98c6b798ccbd4f02d27ce161fef')
+    version('1.8.0-rc1', sha256='ed0395880c32c48a284b115279d27d79ab1ca6fb53a4b97a8d25eba54ec97306')
+    version('1.7.2', sha256='0847943dd65001f3322b00c7dc4e12f56e70e98c6b798ccbd4f02d27ce161fef', preferred=True)
```
Comment (vchuravy) [inline var/spack/repos/builtin/packages/julia/package.py]: We should probably add 1.7.3 xD

File: var/spack/repos/builtin/packages/julia/package.py [hunk 3]
```diff
@@ -75,16 +90,16 @@ class Julia(MakefilePackage):
...
         'https://raw.githubusercontent.com/spack/patches/0b543955683a903d711a3e95ff29a4ce3951ca13/julia/llvm-11.0.1-julia-1.6.patch',
         sha256='8866ee0595272b826b72d173301a2e625855e80680a84af837f1ed6db4657f42'))
     depends_on('llvm', when='^llvm@12.0.1', patches=patch(
         'https://github.com/JuliaLang/llvm-project/compare/fed41342a82f5a3a9201819a82bf7a48313e296b...980d2f60a8524c5546397db9e8bbb7d6ea56c1b7.patch',
         sha256='10cb42f80c2eaad3e9c87cb818b6676f1be26737bdf972c77392d71707386aa4'))
-    depends_on('llvm', when='^llvm@13.0.0', patches=patch(
-        'https://github.com/JuliaLang/llvm-project/compare/d7b669b3a30345cfcdb2fde2af6f48aa4b94845d...6ced34d2b63487a88184c3c468ceda166d10abba.patch',
-        sha256='92f022176ab85ded517a9b7aa04df47e19a5def88f291e0c31100128823166c1'))
+    depends_on('llvm', when='^llvm@13.0.1', patches=patch(
+        'https://github.com/JuliaLang/llvm-project/compare/75e33f71c2dae584b13a7d1186ae0a038ba98838...5393efbd8a4c7555b9f9fdf185c486c6b05f0c19.patch',
```
Comment (vchuravy) [inline var/spack/repos/builtin/packages/julia/package.py]: Needs to be https://github.com/JuliaLang/llvm-project/releases/tag/julia-13.0.1-1 which I am in process of backporting.

It includes a fix for Summit.

**Comment by vchuravy** [comment]
Needs a rebase after #31196

File: var/spack/repos/builtin/packages/libuv-julia/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/libuv-julia/package.py]: it's the best we have right now ;p

**Review by vchuravy — DISMISSED** [review]

File: var/spack/repos/builtin/packages/libuv-julia/package.py (hunk 1 above)
Comment (giordano) [reply var/spack/repos/builtin/packages/libuv-julia/package.py]: I think a `commit` reference with a [git source](https://spack.readthedocs.io/en/latest/packaging_guide.html#git) might be more accurate?

Comment (trws) [reply var/spack/repos/builtin/packages/libuv-julia/package.py]: This isn't referencing a branch, it's referencing a released tarball.  If a 1.44.2 release is made then we will add the version and hash, if the 1.44.1 tarball changes (which should not happen) we will find out when the hash breaks.

Comment (haampie) [reply var/spack/repos/builtin/packages/libuv-julia/package.py]: > This isn't referencing a branch,

`refs/heads/julia-uv2-1.44.1` is a branch.

It seems `https://github.com/JuliaLang/libuv/archive/1b2d16477fe1142adea952168d828a066e03ee4c.tar.gz` works too

Comment (trws) [reply var/spack/repos/builtin/packages/libuv-julia/package.py]: Ah, dangit, my apologies yes it is.  Still, the hash would tell us if and when that branch changes.  Using the commit hash URL would be less likely to break though.

**Review by alalazo — APPROVED** [review]
@haampie Feel free to merge if you're not waiting for further comments from @vchuravy

**Pull request merged**